did_jwk/src/producer.rs

```diff
@@ -1,96 +1,134 @@
-use identity_iota::core::Object;
-use identity_iota::verification::VerificationMethod;
-use identity_iota::{core::ToJson, did::CoreDID, document::CoreDocument, storage::KeyId};
-use log::info;
-use serde_json::json;
-use shared::JwkStorageWrapper;
+use identity_iota::{
+    core::{FromJson, ToJson},
+    did::CoreDID,
+    document::CoreDocument,
+    verification::{jwk::Jwk, jws::JwsAlgorithm, VerificationMethod},
+};
+use log::{debug, info};
+use shared::{error::ProducerError, JwkStorageWrapper};
 use ssi_dids::{DIDMethod, Source};
-use std::io::Error;
-use std::io::ErrorKind;
 
 // See specification: "Since did:jwk only contains a single key, the DID URL fragment identifier is always a fixed #0 value."
 const FRAGMENT: &str = "0";
 
-pub async fn produce_did_jwk(storage: JwkStorageWrapper, key_id: &str) -> Result<CoreDocument, Error> {
-    let public_key_jwk = match storage {
-        JwkStorageWrapper::Stronghold(stronghold_storage) => {
-            stronghold_storage.get_public_key(&KeyId::new(key_id)).await.unwrap()
-        }
-        JwkStorageWrapper::PKCS11 => todo!(),
-    };
+pub async fn produce_did_jwk(
+    storage: JwkStorageWrapper,
+    key_id: &str,
+    alg: JwsAlgorithm,
+) -> Result<CoreDocument, ProducerError> {
+    let public_key_jwk = storage.get_public_key(key_id, alg).await?;
 
-    let jwk: ssi_jwk::JWK = serde_json::from_value(public_key_jwk.to_json_value().unwrap()).unwrap();
+    info!("Producing `did:jwk` for key_id `{key_id}` ({alg}) ...");
 
-    info!("Producing did:jwk for key_id=[{:?}] ...", key_id);
+    let jwk: ssi_jwk::JWK = serde_json::from_value(public_key_jwk.clone()).unwrap();
 
     if let Some(did_str) = did_jwk_extern::DIDJWK.generate(&Source::Key(&jwk)) {
-        info!("DID: {:?}", did_str);
+        info!("DID: `{did_str}`");
 
         let controller = CoreDID::parse(did_str).unwrap();
 
-        let verification_method =
-            VerificationMethod::new_from_jwk(controller.clone(), public_key_jwk.clone(), Some(FRAGMENT)).unwrap();
-
-        let mut properties = Object::new();
-        properties.insert(
-            "@context".to_string(),
-            json!([
-                "https://www.w3.org/ns/did/v1",
-                "https://w3id.org/security/suites/ed25519-2020/v1" // TODO: make dynamic
-            ]),
-        );
```

Comment:
Since `did:jwk` is a deterministic method, the DID document itself is created by each party individually and not transferred in any way - even if there was complete nonsense in the local document, the consumer would never receive it. We can just ignore it.